### PR TITLE
adding closing-time to dropdownMenu

### PR DESCRIPTION
**Pull request by jongear**
adding closing-time to dropdownMenu

### CODE REVIEW

File: dist/angular-foundation-no-tpls.js [hunk 1]
```diff
@@ -287,63 +288,72 @@ angular.module('mm.foundation.dropdownMenu', []).directive('dropdownMenu', ['$co
...
 
             var vm = this;
-            // $element is-dropdown-submenu-parent
         }]
     };
-}]).directive('li', function () {
-    return {
+}]).directive('li', ['$timeout', dropdownMenuItem]);
+
+dropdownMenuItem.$inject = ['$timeout'];
```
Comment (circlingthesun) [inline dist/angular-foundation-no-tpls.js]: We're using angular-annotate so there is no need for setting $inject manually. Could you instead add `'ngInject';` on the first line of the body of `dropdownMenuItem`?

File: src/dropdownMenu/dropdownMenu.js [hunk 2]
```diff
@@ -4,70 +4,79 @@ angular.module('mm.foundation.dropdownMenu', [])
...
         controller: function($scope, $element) {
             'ngInject';
             var vm = this;
-            // $element is-dropdown-submenu-parent
         }
     };
 })
-.directive('li', () => {
-    return {
+.directive('li', ['$timeout', dropdownMenuItem]);
```
Comment (circlingthesun) [inline src/dropdownMenu/dropdownMenu.js]: We're using angular-annotate so there is no need for setting $inject manually. Could you instead add 'ngInject'; on the first line of the body of dropdownMenuItem?

**Comment by circlingthesun** [comment]
Hey @jongear, thanks for the PR. It looks good. If you could you please remove the changes to the build artefacts in `/dist` and address the in-line comment I made above, we should be good to go.

**Comment by jongear** [comment]
Hey @circlingthesun I corrected the outstanding issue.

**Comment by circlingthesun** [comment]
When your mouse leaves the submenu momentarily and then returns, the timeout triggers despite it being over the submenu. The mouseleave event should probably save the promise returned by $timeout, and cancel the callback if a subsequent mouseenter occurs within the delay window. Thoughts?

**Comment by jongear** [comment]
Completely agree. Great point

**Comment by jongear** [comment]
added the promise cancellation for when a user mouses back into the submenu

**Comment by circlingthesun** [comment]
Awesome! Thanks a stack :+1:

**Pull request merged**